src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/RemoteAppServerOptions.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters;
+
+/// <summary>
+/// Options for connecting to a remote app.
+/// </summary>
+public class RemoteAppServerOptions
+{
+    /// <summary>
+    /// Gets or sets the header used to store the API key
+    /// </summary>
+    [Required]
+    public string ApiKeyHeader { get; set; } = RemoteConstants.ApiKeyHeaderName;
+
+    /// <summary>
+    /// Gets or sets an API key used to secure the endpoint
+    /// </summary>
+    [Required]
+    public string ApiKey { get; set; } = null!;
```

Comment:
Can we leave the common parts of the options in a shared library and have the client/server-specific options derive from them? Two days ago, I would have liked this approach for how it separated ASP.NET and ASP.NET Core scenarios but now that I'm adding validation logic for remote app options that will be common to both (#169), it occurs to me that there's value in having the shared portion of the options actually be shared. The two already have a shared dependency of RemoteConstants, so I don't feel too bad having them have a common base class.

RemoteAppServerOptions probably won't add anything to the base class, but that seems fine.

Reply:
Chatted offline about using data validations for the new validation code so it'll be an easy thing to keep separate.